src/linear_solver_routines/inner_block_solving_routines/Gentlemans_House.jl

```diff
@@ -0,0 +1,147 @@
+"""
+    GentData{S, M<:Matrix{S}, V<:Vector{S}}
+
+This is a preallocated data structure for solving least squares problems with 
+Gentleman's algorithm. This data structure is not exported and thus is not 
+designed to be manipulated by the user.
+# Fields
+- `A::M`, The matrix to be solved.
+- `B::M`, The matrix that rows of `A` are brought to have QR decomposition
+applied.
+- `R::UpperTriangular`, The upper triangular part of `B`.
+- `tab::SubArray`, the last column of `B` where the constant vector entries corresponding
+to the rows of `A` that were brought to `B` are stored.
+- `v::V`, a buffer vector.
+- `bsize::Int64`, the number of rows transported to `B` at each iteration.
+
+Calling `Gent(A, bsize)` will allocate a `B` matrix with the n + 1 columns and 
+n + bsize + 1 rows where n is the number of columns of A. It also allocates a 
+buffer vector `v` with n + 1 entries. Finally, it takes a view of the upper
+triangular part of `B[1:n, 1:n]` and a view of the last column of `B`.
+
+Miller, Alan J. "Algorithm AS 274: Least squares routines to supplement those of Gentleman." Applied Statistics (1992): 458-478.
+"""
+mutable struct GentData{S, M<:Matrix{S}, V<:Vector{S}}
+    A::M 
+    B::M
+    R::UpperTriangular
+    tab::SubArray
+    v::V
+    bsize::Int64
+end
+
+Base.eltype(::GentData{S, M, V}) where {S, M, V} = S
+
+# Constructor for Gent data structure
+function Gent(A::AbstractMatrix, bsize::Int64)
```

Comment:
@npritch928 better to have all functions in lower case for consistency

Reply:
This actually is supposed to be a constructor GentData. I did change the capitalization on the other functions though.